### ISSUE TITLE
Add response flags to requests metrics

### ISSUE THREAD

**Issue by billybanfield** [envoyproxy/envoy#6431]
*Title*: Add response flags to request metrics

*Description*:
Request metrics take the form of `envoy_cluster_upstream_rq{envoy_response_code="503",envoy_cluster_name="<cluster_name>"}` in the `/stats?format=prometheus` endpoint. In the access logs, each log line [can be configured to include the response_flags](https://www.envoyproxy.io/docs/envoy/latest/configuration/access_log#command-operators) associated with Envoy's response. Adding the `response_flags` that are present in the access log to these metrics would help disambiguate this data and help more quickly determine root causes.

**Comment by mattklein123** [comment]
This needs some thinking to avoid metric explosion but definitely is possible.

**Comment by billybanfield** [comment]
Hey @mattklein123 ! I'd love to re-open this because this would be a very helpful addition for us. If this is feasible, happy to try my hand at implementing (although I'm not c++ expert) if someone can point me in the right direction here.

**Comment by mattklein123** [comment]
Reopening and marking help wanted. A few notes:
1) We would need to make this configurable to avoid metric explosion.
2) It's going to involve plumbing into https://github.com/envoyproxy/envoy/blob/master/source/common/http/codes.cc as well as setting up the right tag extractors.

cc @jmarantz for further thoughts.

**Comment by jmarantz** [comment]
Yeah I think that's right; gate the addition of these response-code metric variants by a command-line flag, default off.

I can help review if someone wants to take on building it.

**Comment by hochuenw-dd** [comment]
Hi, @mattklein123 @jmarantz @williambanfield, wondering if this feature request still makes sense to the team. I would love to contribute to this as well.

**Comment by sekar-saravanan** [comment]
+1 adding response_flag to the metric will be really helpful during the classification of the root cause of 5xx araised, if any.
